avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -935,7 +940,21 @@ private static Object maybeWrapSchemaless(Schema schema, Object value, String ty
           int precision = precisionString == null ? CONNECT_AVRO_DECIMAL_PRECISION_DEFAULT :
               Integer.parseInt(precisionString);
           int scale = scaleString == null ? 0 : Integer.parseInt(scaleString);
-          org.apache.avro.LogicalTypes.decimal(precision, scale).addToSchema(baseSchema);
+          if (scale < 0 || scale > precision) {
+            log.debug(
+                "Scale and precision of {} and {} cannot be serialized as native Avro logical " 
+                    + "decimal type; reverting to legacy serialization method",
+                scale,
+                precision
+            );
+            // We cannot use the Avro Java library's support for the decimal logical type when the
+            // scale is either negative or greater than the precision as this violates the Avro spec
+            // and causes the Avro library to throw an exception, so we fall back in this case to
+            // using the legacy method for encoding decimal logical type information.
+            forceLegacyDecimal = true;
```

Comment:
Why would we not just add `CONNECT_AVRO_DECIMAL_PRECISION_PROP` as a property to the schema here? Then we wouldn't need the `forceLegacyDecimal` variable and would not need to modify line 963 (original). Or am I missing something?

Reply:
Ran into this after adding the test, turns out the schema parameters map is (at least in some cases) an unmodifiable collection.

Reply:
See https://github.com/apache/kafka/blob/71113b4aac43a3a6f13625e25171ef8f2233fe7d/connect/api/src/main/java/org/apache/kafka/connect/data/SchemaBuilder.java#L186-L189

Reply:
Seems worthy of adding to the comment. I'd lose a lot of time trying to track that down.

Reply:
Ack, added